executor/analyze.go

```diff
@@ -112,6 +113,14 @@ type analyzeTask struct {
 }
 
 func (e *AnalyzeExec) analyzeWorker(taskCh <-chan *analyzeTask, resultCh chan<- statistics.AnalyzeResult) {
+	defer func() {
```

Comment:
How about https://github.com/Shopify/sarama/blob/d49f403db12a8db7ec1ca748519b7708e9a621b4/utils.go#L35
then https://github.com/Shopify/sarama/blob/f4466a0fac66c9d21d58f91158ca758c512a80f5/async_producer.go#L100

I found it's a good practice to keep use `go withRecover(f)` instead of `go f()` whenever we want to `go xx()`, and we are no need and no easy miss recovery call in new goroutine